ably/realtime/connection.py

```diff
@@ -131,13 +132,31 @@ def __init__(self, realtime, initial_state):
         self.__ping_future = None
         self.__timeout_in_secs = self.options.realtime_request_timeout / 1000
         self.transport: WebSocketTransport | None = None
+        self.__ttl_task = None
+        self.__retry_task = None
+        self.__connection_details = None
+        self.__in_suspended_state = False
```

Comment:
what's the reason for this field? does it mean something different from `self.__state == ConnectionState.SUSPENDED`?

Reply:
I had the field to track when the connection has entered the suspended state. We want the suspended connection to be in CONNECTING state when a connection attempt is being made and back to SUSPENDED if the attempt fails without having to go to the `DISCONNECTED` state. At the point of re-attempting  a suspend connection, the state would have changed to `CONNECTING`, so `self.__state == ConnectionState.SUSPENDED` will no longer hold

Reply:
Ah I see, thanks for explaining 🙂 In that case I think this name is _slightly_ misleading because you can have a situation where `self.__state != ConnectionState.SUSPENDED and self.__in_suspended_state = True`. I think it would be better to have a `self.__fail_state` field which starts off as `ConnectionState.DISCONNECTED` but changes to `ConnectionState.SUSPENDED` when `connectionStateTtl` has elapsed

Reply:
`fail_state` should represent the desired connection state when a connection attempt has failed, so:
- it should start off as `DISCONNECTED`
- should be changed to `SUSPENDED` whenever the connection becomes `SUSPENDED`
- should be changed back to `DISCONNECTED` when connection is established again

Reply:
Yes, I did just that